uint8[2] size
---
bool succeed
